src/main/java/mmosii/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+import mmosii.bookstore.mapper.OrderMapper;
+import mmosii.bookstore.model.CartItem;
+import mmosii.bookstore.model.Order;
+import mmosii.bookstore.model.OrderItem;
+import mmosii.bookstore.model.ShoppingCart;
+import mmosii.bookstore.model.User;
+import mmosii.bookstore.repository.order.OrderItemRepository;
+import mmosii.bookstore.repository.order.OrderRepository;
+import mmosii.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import mmosii.bookstore.repository.user.UserRepository;
+import mmosii.bookstore.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    @Transactional
+    public OrderDto addOrder(CreateOrderRequestDto requestDto) {
+        User user = findUser();
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(()
+                -> new EntityNotFoundException("Can't find shopping cart by users id: "
+                                               + user.getId()));
+        Order order = createNewOrder(user, requestDto);
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        Set<OrderItem> orderItems = new HashSet<>();
+        for (CartItem cartItem : cartItems) {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setOrder(order);
+            orderItem.setPrice(orderItem.getBook().getPrice());
+            orderItems.add(orderItemRepository.save(orderItem));
+            order.setTotal(order.getTotal()
+                    .add(orderItem.getPrice()
+                            .multiply(BigDecimal.valueOf(orderItem.getQuantity()))));
+        }
+        shoppingCartRepository.delete(shoppingCart);
+        order.setOrderItems(orderItems);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderDto> getOrders(Pageable pageable) {
+        return orderMapper.toDtoList(orderRepository.findByUserId(findUser().getId(), pageable));
+    }
+
+    @Transactional
+    @Override
+    public OrderDto updateOrder(Long id, UpdateStatusRequestDto requestDto) {
+        Order order = orderRepository.getReferenceById(id);
+        order.setStatus(Order.Status.valueOf(requestDto.status()));
+        return orderMapper.toDto(orderRepository.save(order));
```

Comment:
there is no need to call `save` since you have an active session

and please verify that `getReference` here doesn't break mapper logic